js/tweek-client/src/TweekManagementClient/TweekManagementClient.ts

```diff
@@ -233,6 +234,47 @@ export default class TweekManagementClient implements ITweekManagementClient {
     return this._fetch(url).then(toJson);
   }
 
+  getHooks(keyPathFilter?: string): Promise<Hook[]> {
+    let url = `${this.config.baseServiceUrl}/api/v2/hooks`;
+
+    if (keyPathFilter) {
+      const queryParamsObject: InputParams = { keyPathFilter };
+      const queryString = toQueryString(queryParamsObject);
+      url += `/${queryString}`;
+    }
+
+    return this._fetch(url).then(toJson);
+  }
+
+  createHook(hookData: { keyPath: string; type: string; url: string }): Promise<Hook> {
+    const requestUrl = `${this.config.baseServiceUrl}/api/v2/hooks`;
+    const config = {
+      method: 'POST',
+      headers: jsonHeaders,
+      body: JSON.stringify(hookData),
+    };
+
+    return this._fetch(requestUrl, config).then(toJson);
+  }
+
+  updateHook({ id, ...hookData }: Hook): Promise<void> {
+    const requestUrl = `${this.config.baseServiceUrl}/api/v2/hooks/${id}`;
+    const config = {
+      method: 'PUT',
+      headers: jsonHeaders,
+      body: JSON.stringify(hookData),
+    };
+
+    return this._fetch(requestUrl, config).then(noop);
+  }
+
+  deleteHook({ id }: { id: string }): Promise<void> {
```

Comment:
why not just `id: string` parameter?

Reply:
I wanted to allow sending the Hook object itself as a parameter to this function